filesystem/fat32/file.go

```diff
@@ -154,6 +156,9 @@ func (fl *File) Write(p []byte) (int, error) {
 		file.WriteAt(p[totalWritten:totalWritten+toWrite], int64(offset)+fs.start)
 		totalWritten += toWrite
 	}
+
+	fl.offset = fl.offset + int64(totalWritten)
```

Comment:
Personally, I think that Unix should have separate offsets for read and write. But we are not about to change 40+ years of history here, and we should stick to the standard, so this makes sense.